### ISSUE TITLE
Tuple4 is not public despite being part of the API

### ISSUE THREAD

**Issue by rkraneis** [jlink/jqwik#9]
## Testing Problem
To circumvent the limitation of only being able to combine 4 arbitraries, I'd like to group them into tuples, but Tuple4 is not public.

## Suggested Solution

Make Tuple4 public.

## Discussion

Ideally, there would also be some kind of ArbitraryBuilder, as writing them from scratch is quite cumbersome and using my workaround is also not great:
```
    @Provide()
    Arbitrary<ComplexObject> complexObjectGen() {
        Arbitrary<Foo> foos1 = fooGen();
        Arbitrary<Foo> foos2 = fooGen();
        Arbitrary<Foo> foos3 = fooGen();
        Arbitrary<Foo> foos4 = fooGen();
        Arbitrary<Bar> bars1 = barGen();
        Arbitrary<Bar> bars2 = barGen();
        Arbitrary<Bar> bars3 = barGen();
        Arbitrary<Bar> bars4 = barGen();
        
        Arbitrary<Tuple4<Foo, Foo, Bar, Bar>> combine1 = Combinators.combine(foos1, foos2, bars1, bars2).as((f1, f2, b1, b2) -> tuple(f1, f2, b1, b2));
        Arbitrary<Tuple4<Foo, Foo, Bar, Bar>> combine2 = Combinators.combine(foos3, foos4, bars3, bars4).as((f3, f4, b3, b4) -> tuple(f3, f4, b3, b4));
        
        return Combinators.combine(combine1, combine2).as((c1, c2) -> ComplexObject.create(c1.v1, c1.v2, c1.v3, c1.v4, c2.v1, c2.v2, c2.v3, c2.v4));
    }
```

**Comment by jlink** [comment]
Tuple4 should definitely be public. I’ll fix it. 

As for your original reason: How many arbitraries would you like to combine?

**Comment by rkraneis** [comment]
I guess there should be a middle ground between Scalas 22 and the current 4. Vavr opts for 8, as does functionaljava. I think for everything larger a workaround like mine is probably OK-ish. It can even be written somewhat shorter if there are small lambdas for the initial combinators.

**Comment by jlink** [comment]
Made Tuple4 public in https://github.com/jlink/jqwik/commit/1fb691695ee2aedb8e57ef9fd2661ce2094cb1c2

**Comment by jlink** [comment]
Added possibility to combine up to 8 arbitraries in https://github.com/jlink/jqwik/commit/d8dd9eef19944637fd5196984788aed01a1ec8ab

**Comment by jlink** [comment]
Available in 0.8.4-SNAPSHOT. Will be released in version 0.8.4.
